### PR TITLE
Setup CI and run forge fmt and prettier

### PR DESCRIPTION
**Pull request by alcueca**
Secrets have been set to enable this

Remember to run:
```
pnpm prettier --check **/*.md --ignore-path=.prettierignore -w
forge fmt
```

### CODE REVIEW

File: foundry.toml [hunk 1]
```diff
@@ -1,13 +1,58 @@
...
+  gas_reports = ["*"]
+  libs = ["lib"]
+  optimizer = true
+  optimizer_runs = 10_000
+  out = "out"
+  script = "script"
+  solc = "0.8.20"
+  src = "src"
+  test = "test"
+  evm_version = 'london'
```
Comment (totomanov) [inline foundry.toml]: make evm_version "paris" if we're targeting evms or "cancun" if we're targeting ethereum

File: foundry.toml [hunk 2]
```diff
@@ -1,13 +1,58 @@
...
+  bytecode_hash = "none"
+  cbor_metadata = false
+  fuzz = { runs = 1_000 }
+  gas_reports = ["*"]
+  libs = ["lib"]
+  optimizer = true
+  optimizer_runs = 10_000
+  out = "out"
+  script = "script"
+  solc = "0.8.20"
```
Comment (totomanov) [inline foundry.toml]: we can use the latest solc but still choose if we wnat backwards compatibility through evm_version

Comment (totomanov) [reply foundry.toml]: the key `solc` id deprecated afaik, use `solc_version` now

File: foundry.toml [hunk 3]
```diff
@@ -1,13 +1,58 @@
+# Full reference https://github.com/foundry-rs/foundry/tree/master/config
+
 [profile.default]
-src = "src"
-out = "out"
-libs = ["lib"]
-verbosity = 3
-solc_version = '0.8.20'
+  auto_detect_solc = false
```
Comment (totomanov) [inline foundry.toml]: ignored if `solc_version` is set

File: foundry.toml [hunk 4]
```diff
@@ -1,13 +1,58 @@
...
+[etherscan]
+  arbitrum_one = { key = "${API_KEY_ARBISCAN}", chain = 42161 }
+  # avalanche = { key = "${API_KEY_SNOWTRACE}", chain = 43114 }
+  # bnb_smart_chain = { key = "${ API_KEY_BSCSCAN}", chain = 56 }
+  gnosis_chain = { key = "${API_KEY_GNOSISSCAN}", chain = 100 }
+  # goerli = { key = "${API_KEY_ETHERSCAN}", chain = 5 }
+  mainnet = { key = "${API_KEY_ETHERSCAN}", chain = 1 }
+  optimism = { key = "${API_KEY_OPTIMISTIC_ETHERSCAN}", chain = 10 }
+  polygon = { key = "${API_KEY_POLYGONSCAN}", chain = 137 }
+  # sepolia = { key = "${API_KEY_ETHERSCAN}", chain = 58008 }
```
Comment (totomanov) [inline foundry.toml]: sepolia is 11155111

also why support a dozen evms?

File: foundry.toml [hunk 5]
```diff
@@ -1,13 +1,58 @@
...
+
+[fmt]
```
Comment (totomanov) [inline foundry.toml:35]: unless you have an explicit preference for some of the `fmt` settings, I'd keep everything default

File: foundry.toml (hunk 4 above)
Comment (alcueca) [reply foundry.toml]: I got the toml file from elsewhere, and the supported evems were already there.

But anyway, I think that this project will be useful for all evms.

**Pull request merged**